common/src/lib.rs

```diff
@@ -183,8 +183,14 @@ pub trait BlockLimiter {
 #[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, TypeInfo)]
 pub enum Program {
```

Comment:
As I see now, each field contains bn of insertion in storage, so atm looks better for me to come back to previous Program enum structure in order to change type stored on-chain: map KEY `ProgramId` - VALUE `(Program, BlockNumberFor<T>)`

Reply:
but these changes also required js changes @osipov-mit

Reply:
I think it's worth introducing a separate storage for the bn (all the data related to the charges for resources). One of reasons for that - we don't need bn every time we read the program storage. Even most of time we don't need it. So better to separate it to read less bytes.

Reply:
> As I see now, each field contains bn of insertion in storage, so atm looks better for me to come back to previous Program enum structure in order to change type stored on-chain: map KEY `ProgramId` - VALUE `(Program, BlockNumberFor<T>)`

Since "block_number" is a progeram attribute it's logically to add this parameter inside of the struct. Seems that readability is better this way. What might be wrong with it?

Reply:
Frankly speaking introducing a separate storage for program block number seems better way.

Reply:
Done